nrf-softdevice-macro/src/lib.rs

```diff
@@ -226,6 +227,35 @@ pub fn gatt_server(args: TokenStream, item: TokenStream) -> TokenStream {
                 }
             ));
         }
+        if indicate {
+            let case_enabled = format_ident!("{}IndicationsEnabled", name_pascal);
+            let case_disabled = format_ident!("{}IndicationsDisabled", name_pascal);
+
+            code_impl.extend(quote_spanned!(ch.span=>
+                fn #indicate_fn(
+                    &self,
+                    conn: &#ble::Connection,
+                    val: #ty,
+                ) -> Result<(), #ble::gatt_server::IndicateValueError> {
+                    let buf = #ty_as_val::to_gatt(&val);
+                    #ble::gatt_server::indicate_value(conn, self.#value_handle, buf)
+                }
+            ));
+
+            code_event_enum.extend(quote_spanned!(ch.span=>
+                #case_enabled,
+                #case_disabled,
+            ));
+            code_on_write.extend(quote_spanned!(ch.span=>
+                if handle == self.#cccd_handle {
+                    if data.len() != 0 && data[0] & 0x02 != 0 {
+                        return Some(#event_enum_name::#case_enabled);
+                    } else {
+                        return Some(#event_enum_name::#case_disabled);
```

Comment:
Should there be a guard preventing specifying both notify and indicate attribute? If not, I think this will cause a IndicationsDisabled event to be missed if notify was enabled as well (the on_write would return NotificationsDisabled earlier in the generated code).

Reply:
Ah, good catch! I think enabling both notifications and indications is a valid use case so I rather think the previous value of `data[0]` needs to be stored.. I'll look into it some more